### DISCUSSION TITLE
what is the relation between opensumi and vs code?

### DISCUSSION THREAD

**Discussion by weibinzhu** [opensumi/core#541] [Q&A] [answered]
My first impression on opensumi is quite similar to vs code's. 
So I wonder whether opensumi was modified from vs code.

**Comment by erha19** [comment, marked as answer]
@weibinzhu 

Firstly, OpenSumi is not forked from VS Code. 

We need to follow some code designs and standard Extension APIs of VS Code cause we hope to be compatible with the extension ecosystem of VS Code at the begging.

We have some notable differences from VS Code：

- We are a framework, but VS Code is an IDE product.
- We build the web frontend with React, but VS Code uses pure JS.
- We support a lot of views custom API on `sumi` API, a developer can use `module` or `extension` to custom all the views you see.

Also， you can see all our code on the repo [opensumi/core](https://github.com/opensumi/core), many of them are implemented independently.
